### PR TITLE
Added load-test framework to repo

### PR DESCRIPTION
**Pull request by jrheizelman**
Copied over Rohit's work on git-on-borg to the repo under the folder load-test-framework. No changes beyond a slight readme change to explain that Flic (Rohit's assigned name) is the load-testing framework to compare Kafka and CPS.

### CODE REVIEW

File: load-test-framework/run.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/python
```
Comment (tmatsuo) [inline load-test-framework/run.py]: I recommend that we adhere to pep8 on github

File: load-test-framework/run.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+BASE_EXEC_PATH = 'java -Xmx20g -jar target/flic.jar '
+
+
+def kafka_pub_load_test():
+  # Runs a test which publishes to Kafka
+  args = BASE_EXEC_PATH + \
+  '--num_messages 10000000 \
+  --message_size 10 \
+  --topics [YOUR TOPIC HERE] \
```
Comment (tmatsuo) [inline load-test-framework/run.py]: Are users supposed to rewrite this file? Why not make variables args, or envvars?

File: load-test-framework/README.md [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+2.  Make the jar that contains the connector:
+
+    `mvn package`
+
+The resulting jar is at target/flic.jar.
+
+### Pre-Running Steps
+
+1.  Regardless of whether you are running on Google Cloud Platform or not, you
+    need to create a project and create service key that allows you access to
```
Comment (kamalaboulhosn) [inline load-test-framework/README.md]: "create a service key"

File: load-test-framework/README.md [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+    export this environment variable as part of your shell startup file).
+
+    `export GOOGLE_APPLICATION_CREDENTIALS=/path/to/key/file`
+
+### Important Notes
+
+It is important to first point out that we are providing a framework for load
+testing Cloud Pub/Sub. While it is possible to publish and consume messages to
+Kafka from this framework, the client code used to do so is not purposely
+optimized to show off the things Kafka does well. On the other hand, the client
```
Comment (kamalaboulhosn) [inline load-test-framework/README.md]: Better to say "has not yet been optimized," since we plan to do so.

File: load-test-framework/run.py (hunk 2 above)
Comment (kamalaboulhosn) [inline load-test-framework/run.py]: Yeah, it is strange that the README talks about command-line arguments, but those are not present here in the actual startup script.

File: load-test-framework/src/main/java/com/google/pubsub/flic/Driver.java [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+      JCommander jCommander = new JCommander(baseArgs);
+      jCommander.addCommand(CPSArguments.COMMAND, cpsArgs);
+      jCommander.addCommand(KafkaArguments.COMMAND, kafkaArgs);
+      jCommander.addCommand(CompareArguments.COMMAND, dataComparisonArgs);
+      jCommander.parse(args);
+      if (jCommander.getParsedCommand() == null) {
+        if (baseArgs.isHelp()) {
+          jCommander.usage();
+        }
+        System.exit(0);
```
Comment (kamalaboulhosn) [inline load-test-framework/src/main/java/com/google/pubsub/flic/Driver.java]: All of these System.exit(0) in main should just be returns.

File: load-test-framework/src/main/java/com/google/pubsub/flic/common/MessagePacketProto.java [hunk 6]
```diff
@@ -0,0 +1,745 @@
+// Generated by the protocol buffer compiler.  DO NOT EDIT!
```
Comment (kamalaboulhosn) [inline load-test-framework/src/main/java/com/google/pubsub/flic/common/MessagePacketProto.java]: Do we need to include the generated file? I would think the Maven compilation would take care of generating it on the fly.

File: load-test-framework/src/main/java/com/google/pubsub/flic/argumentparsing/CPSArguments.java [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+
+  public String getProject() {
+    return project;
+  }
+
+  public int getBatchSize() {
+    return batchSize;
+  }
+
+
```
Comment (kamalaboulhosn) [inline load-test-framework/src/main/java/com/google/pubsub/flic/argumentparsing/CPSArguments.java]: Extra newline.

File: load-test-framework/src/main/java/com/google/pubsub/flic/cps/CPSPublishingTask.java [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+    PubsubMessage.Builder messageBuilder = PubsubMessage.newBuilder();
+    PublishRequest.Builder requestBuilder = PublishRequest.newBuilder();
+    // Warm up all threads in the thread pool.
+    callbackExecutor.prestartAllCoreThreads();
+    // Get the overall start time.
+    long start = System.currentTimeMillis();
+    while (messageNo.intValue() <= args.getNumMessages() && !failureFlag.get()) {
+      List<PubsubMessage> batch = new ArrayList<>();
+      // Keep track of the total # of bytes in this batch and the size of the batch.
+      AtomicLong batchBytes = new AtomicLong(0);
```
Comment (kamalaboulhosn) [inline load-test-framework/src/main/java/com/google/pubsub/flic/cps/CPSPublishingTask.java]: Do batchBytes and counter need to be atomic? It doesn't look like we are doing anything with them in a multithreaded context.

Comment (kamalaboulhosn) [inline load-test-framework/src/main/java/com/google/pubsub/flic/cps/CPSPublishingTask.java]: Do batchBytes and counter need to be atomic? I see we use them later in a future, but by that point, we essentially hand ownership over to the future and don't make any more updates to the values.

File: load-test-framework/README.md (hunk 3 above)
Comment (jrheizelman) [inline load-test-framework/README.md]: Fixed

File: load-test-framework/README.md (hunk 4 above)
Comment (jrheizelman) [inline load-test-framework/README.md]: Fixed.

File: load-test-framework/src/main/java/com/google/pubsub/flic/Driver.java (hunk 5 above)
Comment (jrheizelman) [inline load-test-framework/src/main/java/com/google/pubsub/flic/Driver.java]: Fixed

File: load-test-framework/src/main/java/com/google/pubsub/flic/argumentparsing/CPSArguments.java (hunk 7 above)
Comment (jrheizelman) [inline load-test-framework/src/main/java/com/google/pubsub/flic/argumentparsing/CPSArguments.java]: Fixed

File: load-test-framework/run.py (hunk 2 above)
Comment (jrheizelman) [inline load-test-framework/run.py]: I removed them, and they can always be restored if it turns out we needed them.

File: load-test-framework/run.py (hunk 1 above)
Comment (jrheizelman) [inline load-test-framework/run.py]: Installed the pep8 check, and the code passed, so I believe this is fixed. In the future, will use this style.

File: load-test-framework/src/main/java/com/google/pubsub/flic/Driver.java [hunk 9]
```diff
@@ -130,7 +130,7 @@ public static void main(String[] args) {
       }
     } catch (Exception e) {
       log.error("An error occurred...", e);
-      System.exit(1);
+      return;
```
Comment (kamalaboulhosn) [inline load-test-framework/src/main/java/com/google/pubsub/flic/Driver.java]: This should still be an exit since the exit code is nonzero.

Comment (jrheizelman) [inline load-test-framework/src/main/java/com/google/pubsub/flic/Driver.java]: Fixed

**Pull request merged**